scarlet/wavelet.py

```diff
@@ -0,0 +1,300 @@
+        else:
+            self._image = image
+        # Original shape of the image
+        self._image_shape = self._image.shape
+        # Can be initialised by a starlet transform to perform inverse transforms
+        if starlet is None:
+            if direct == True:
+                self._starlet = self.direct_transform()
+            else:
+                self._starlet = self.transform()
+        else:
+            if len(np.shape(starlet)) == 3:
+                self._starlet = starlet[np.newaxis, :, :, :]
+            else:
+                self._starlet = starlet
+
+        if self.seed is None:
+            self.seed = mk_starlet(self._starlet_shape)
+        self._norm = np.sqrt(np.sum(self.seed ** 2, axis=(-2, -1)))
+
+    @property
+    def image(self):
+        """The real space image"""
+        return self._image
+
+    @property
+    def norm(self):
+        """The norm of the seed wavelet in each wavelet level (not in coarse wavelet)"""
+        return self._norm
+
+
+    @property
+    def starlet(self):
+        """The real space image"""
+        return fft._centered(self._starlet,
+                             [self._image_shape[0], self._starlet_shape[-3], *self._image_shape[-2:]])
+
+    @property
+    def shape(self):
+        """The shape of the real space image"""
+        return self._image.shape
+
+    @staticmethod
+    def from_starlet(starlet, shape = None, starlet_shape = None):
+        """ Creates a Starlet object from its transform and uses the inverse transform to create the image
+
+        Parameters
+        ----------
+        starlet: array
+            The starlet transform of an array to inverse transform.
+        shape: tuple
+            the expected shape of the untransformed image without padding
+
+        Returns
+        -------
+        Starlet: Starlet object
+            the starlet object initialised with the image that corresponds to the inverse transfform of `starlet`
+        """
+        # Shape of the image to reconstruct
+        if (shape is not None) and (starlet_shape is not None):
+            if not ((starlet.shape[-2:] is shape[-2:]) ^ (starlet.shape[-2:] is starlet_shape[-2:])):
+                raise InputError('Either shape or starlet should have the same shape as the starlet.')
+        if shape is None:
+            shape = [*np.shape(starlet)[:-3],*np.shape(starlet)[-2:]]
+        if (starlet_shape == None):
+            starlet_shape = get_starlet_shape(shape)
+        starlet_shape[0] = starlet.shape[0]
+        if len(starlet.shape) >3:
+            rec = []
+            for star in starlet:
+                rec.append(fft._centered(iuwt(star), shape[-2:]))
+            return Starlet(np.array(rec), starlet = starlet)
+
+        return Starlet(fft._centered(iuwt(starlet), shape), starlet = starlet)
+
+
+
+    def transform(self):
+        """ Performs the wavelet transform of an image by comvolution with the seed wavelet and caches the seed
```

Comment:
typo: `comvolution`. I'd also expand the docstring. For instance, it's not clear what image the transform acts on because the method just has `self` as argument.

The caching of the seed should explained similarly in the expanded docstring

Reply:
Same for the return value. Please use the syntax
```
Returns
-------
...
```

Reply:
Done, please check